modules/abstract-lightning/src/lightning/lightningUtils.ts

```diff
@@ -206,3 +206,32 @@ export function deriveLightningServiceSharedSecret(coinName: 'lnbtc' | 'tlnbtc',
   const userAuthHdNode = utxolib.bip32.fromBase58(userAuthXprv);
   return sdkcore.getSharedSecret(userAuthHdNode, publicKey);
 }
+
+/**
+ * Derives the shared secret for the middleware using the user's auth extended private key and the middleware's public key.
+ * This is used for secure communication between the middleware and the user.
+ */
+export function deriveMiddlewareSharedSecret(coinName: 'lnbtc' | 'tlnbtc', userXprv: string): Buffer {
+  const publicKey = Buffer.from(getStaticsLightningNetwork(coinName).middlewarePubKey, 'hex');
+  const userAuthHdNode = utxolib.bip32.fromBase58(userXprv);
+  return sdkcore.getSharedSecret(userAuthHdNode, publicKey);
+}
+
+/**
+ * Derives the shared secret for TAT service using the user's private key and the TAT public key.
+ * This is used for secure communication with the TAT service and the user.
+ */
+export function deriveTatSharedSecret(coinName: 'lnbtc' | 'tlnbtc', userXprv: string): Buffer {
+  const publicKey = Buffer.from(getStaticsLightningNetwork(coinName).tatPubKey, 'hex');
+  const userAuthHdNode = utxolib.bip32.fromBase58(userXprv);
+  return sdkcore.getSharedSecret(userAuthHdNode, publicKey);
+}
+
+/**
+ * Given a seed, compute a BIP32 derivation index.
+ * 0 <= index < 4294967295 (largest 4 byte number)
+ * @param seed
+ */
+export function computeBip32DerivationIndexFromSeed(seed: string): number {
```

Comment:
making `seed` optional and creating local seed will be more flexible and useful?